jupyterhub_singleuser_profiles/service.py

```diff
@@ -37,91 +41,110 @@ def __init__(self, server_url, token, namespace=None, verify_ssl=True):
-            raise
-        except ApiException as e2:
-          _LOGGER.error("Exception when calling CoreV1Api->create_namespaced_config_map: %s\n" % e2)
 
-  def delete_resource_by_service_label(self, label_value):
-    api_response = self.k8s_api_instance.list_namespaced_config_map(self.namespace, label_selector="%s=%s" % (_SERVICE_LABEL, escape(label_value)))
-    for item in api_response.to_dict()['items']:
+  def submit_resource(self, processed_template, return_paths, user):
+    _LOGGER.debug("Submitting resource")
+    client_wrapper = self.os_client.resources.get(api_version=processed_template['apiVersion'], kind=processed_template['kind'])
+    try:
+      ref_cm = self.get_service_reference_config_map(user)
+      processed_template['metadata']['ownerReferences'] = [{
+            'kind' : ref_cm['kind'],
+            'apiVersion' : ref_cm['apiVersion'],
+            'name' : ref_cm['metadata']['name'],
+            'uid' : ref_cm['metadata']['uid']
+          }]
+      response = client_wrapper.create(body=processed_template, namespace=self.namespace)
+      result = self._get_data_from_response(response, return_paths)
+      return result
+    except ApiException as e:
       try:
-        self.k8s_api_instance.delete_namespaced_config_map(item["metadata"]["name"], self.namespace, client.V1DeleteOptions())
-      except ApiException as e:
-        _LOGGER.error("Exception when calling CoreV1Api->delete_namespaced_config_map: %s\n" % e)
+        if e.status == 409:
+          original_resource = client_wrapper.get(name=processed_template['metadata']['name'], namespace=self.namespace)
+          original_dict = original_resource.to_dict()
+          # Added because of https://github.com/kubernetes/kubernetes/issues/70674#issuecomment-438569688
+          processed_template['metadata']['resourceVersion'] = original_dict['metadata']['resourceVersion']
+          # Necessary for deletion purposes using garbage collection
+          ref_cm = self.get_service_reference_config_map(user)
+          _LOGGER.debug("Added owner reference %s to %s" % (ref_cm['metadata']['name'], processed_template['metadata']['name']))
+          processed_template['metadata']['ownerReferences'] = [{
+            'kind' : ref_cm['kind'],
+            'apiVersion' : ref_cm['apiVersion'],
+            'name' : ref_cm['metadata']['name'],
+            'uid' : ref_cm['metadata']['uid']
+          }]
+          response = client_wrapper.replace(body=processed_template, namespace=self.namespace)
+          result = self._get_data_from_response(response, return_paths)
+          return result
+        else:
+          _LOGGER.error("Error when trying to submit resource %s: %s\n" % (processed_template['metadata']['name'], e))
+          raise
+      except ApiException as e2:
+        _LOGGER.error("Error when trying to submit resource %s: %s\n" % (processed_template['metadata']['name'], e2))
+
+  def delete_resource_by_user(self, user):
```

Comment:
We are working only with CMs here, so the name should probably be `delete_reference_cm`